packages/squiggle-lang/__tests__/TS/Scalars_test.ts

```diff
@@ -0,0 +1,51 @@
+import { errorValueToString } from "../../src/js/index";
+import { testRun } from "./TestHelpers";
+import * as fc from "fast-check";
+
+describe("Scalar manipulation is well-modeled by javascript math", () => {
+  test("in the case of logarithms (with assignment)", () => {
+    fc.assert(
+      fc.property(fc.float(), (x) => {
+        let squiggleString = `x = log(${x}); x`;
+        let squiggleResult = testRun(squiggleString);
+        if (x == 0) {
+          expect(squiggleResult.value).toEqual({
+            tag: "number",
```

Comment:
Maybe we should export a number constructor? Make this a bit cleaner (non critical)

Reply:
irrelevant due to sufficient revamp.